### ISSUE TITLE
Thingsboard serialize json data issue

### ISSUE THREAD

**Issue by gittimotap** [bblanchon/ArduinoJson#2054]
**Describe the bug**  
https://github.com/thingsboard/thingsboard/issues/10164

**Troubleshooter report**  
Here is the report generated by the [ArduinoJson Troubleshooter](https://arduinojson.org/v7/troubleshooter/):  
[Paste the report here]

**Environment**  
Here is the environment that I used:
* Microcontroller: https://wiki.dfrobot.com/FireBeetle_Board_ESP32_E_SKU_DFR0654
* Core/runtime: ESP32 core for Arduino 2.0.14 [e.g. ESP8266 core for Arduino v3.0.2]
* IDE: Arduino IDE 2.0.3

**Reproduction**  
Here is a small snippet that reproduces the issue.

```c++
JsonDocument doc;

DeserializationError error = deserializeJson(doc, "{\"hello\":\"world\"}");

[insert repro code here]
https://github.com/thingsboard/thingsboard-client-sdk/blob/master/examples/0003-esp8266_esp32_send_data/0003-esp8266_esp32_send_data.ino

**Compiler output**    
If relevant, include the complete compiler output (i.e. not just the line that contains the error.)

**Program output**  
If relevant, include the repro program output.
16:59:57.275 -> .Connected to AP
16:59:58.286 -> Connecting to: (demo.thingsboard.io) with token (xxxxx)
16:59:58.493 -> Sending temperature data...
16:59:58.493 -> [TB] Unable to serialize json data
16:59:58.493 -> Sending humidity data...
16:59:58.493 -> [TB] Unable to serialize json data
16:59:59.496 -> Sending temperature data...
16:59:59.496 -> [TB] Unable to serialize json data
16:59:59.496 -> Sending humidity data...
16:59:59.496 -> [TB] Unable to serialize json data
17:00:00.466 -> Sending temperature data...
17:00:00.466 -> [TB] Unable to serialize json data

Expected output:

```
[insert expected output here]
```

Actual output:

```
[insert actual output here]
```

**Comment by gittimotap** [comment]
6.21.5 works, seems like the next release broke something. Found originally here: https://github.com/thingsboard/thingsboard/issues/10164

**Comment by bblanchon** [comment]
Hi @gittimotap,

I need more information.
Please provide at least a reproduction code.

Best regards,
Benoit

**Comment by gittimotap** [comment]
Hi Benoit
Easiest way is to take for example this https://github.com/thingsboard/thingsboard-client-sdk/blob/master/examples/0003-esp8266_esp32_send_data/0003-esp8266_esp32_send_data.ino and run it on some Arduino board (I tried both https://wiki.dfrobot.com/SKU_DFR0868_Beetle_ESP32_C3 and https://wiki.dfrobot.com/FireBeetle_Board_ESP32_E_SKU_DFR0654), it will give you an error straight away. You can use demo.thingsboard.io, you can have a free account there.

In any case, I tried multiple of the SDK examples and all give the same error: https://github.com/thingsboard/thingsboard-client-sdk/tree/master/examples (ESP32 based, not ESP8266). I always get "[TB] Unable to serialize json data" with anything above 6.21.5.

Regards
Timo

**Comment by bblanchon** [comment]
The workaround is available in [ArduinoJson 7.0.4](https://github.com/bblanchon/ArduinoJson/releases/tag/v7.0.4).
